### PR TITLE
Depreciate ozone api type

### PR DESCRIPTION
**Pull request by hubjac1**

### CODE REVIEW

File: packages/ozone-api/ozone-api-type/src/ozone-api-type/ozone-api-type.ts [hunk 1]
```diff
@@ -27,6 +27,10 @@ export type TypeDescriptorCollection = Map<string, Promise<TypeDescriptor>>
  */
 export class OzoneApiType  {
 
+	constructor() {
+		console.warn('OzoneApiType is depreciate, use ozone-typescript-client instead')
```
Comment (lmancel) [inline packages/ozone-api/ozone-api-type/src/ozone-api-type/ozone-api-type.ts]: deprecated*

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
 import * as Config from 'ozone-config'
 
-import {customElement, property} from 'taktik-polymer-typescript'
-import {Item, Media, Channel, Video} from 'ozone-type';
-import {OzoneMediaUrl, OzonePreviewSize, SizeEnum} from 'ozone-media-url'
-import {OzoneApiType, getOzoneApiType} from 'ozone-api-type'
-import {OzoneItemAction} from '../ozone-item-action/ozone-item-action'
-import  '../ozone-item-action/ozone-item-action'
+import { customElement, property } from 'taktik-polymer-typescript'
+import { Item, Media, Channel, Video } from 'ozone-type'
```
Comment (antoinebaudoux) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Remove unused Media, Channel imports

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 3]
```diff
@@ -2,20 +2,20 @@
...
+import { OzoneMediaUrl, OzonePreviewSize, SizeEnum } from 'ozone-media-url'
```
Comment (antoinebaudoux) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Remove unused SizeEnum import

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 4]
```diff
@@ -2,20 +2,20 @@
...
+import { getDefaultClient } from 'ozone-default-client'
+import { OzoneItemAction } from '../ozone-item-action/ozone-item-action'
```
Comment (antoinebaudoux) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Remove unused OzoneItemAction import

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 5]
```diff
@@ -2,20 +2,20 @@
...
+import '../ozone-item-action/ozone-item-action'
```
Comment (antoinebaudoux) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Remove unused OzoneItemAction import

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 6]
```diff
@@ -35,80 +35,76 @@ import  '../ozone-item-action/ozone-item-action'
...
+		}
+		return placeholder
+	}
+
+	async dataChange(data?: Item) {
+		const config = await Config.OzoneConfig.get()
+		this.set('previewImage', undefined)
+		if (data && data.type) {
+			const typeCache = await getDefaultClient().typeClient().getTypeCache()
+			const isTypeInstanceOf = typeCache.isTypeInstanceOf(data.type, 'media')
```
Comment (antoinebaudoux) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Rename const : isMedia

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 7]
```diff
@@ -35,80 +35,76 @@ import  '../ozone-item-action/ozone-item-action'
...
+			if (isTypeInstanceOf) {
+				const media: Video = data as Video
```
Comment (antoinebaudoux) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Remove unchecked cast

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 8]
```diff
@@ -35,80 +35,76 @@ import  '../ozone-item-action/ozone-item-action'
...
+				const id = media.logo || media.id
+				const ozoneMediaUrl = new OzoneMediaUrl(id as string, config)
```
Comment (antoinebaudoux) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Remove unchecked cast

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 9]
```diff
@@ -97,12 +96,14 @@ export class OzoneItemPreview extends Polymer.Element {
 		this.set('previewImage', undefined)
 		if (data && data.type) {
 			const typeCache = await getDefaultClient().typeClient().getTypeCache()
-			const isTypeInstanceOf = typeCache.isTypeInstanceOf(data.type, 'media')
-			if (isTypeInstanceOf) {
+			const isMedia = typeCache.isTypeInstanceOf(data.type, 'media')
+			if (isMedia) {
 				const media: Video = data as Video
```
Comment (lmancel) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Do not cast an item as Video if we're not certain it is one. We should verify it explicitly

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 10]
```diff
@@ -96,10 +96,9 @@ export class OzoneItemPreview extends Polymer.Element {
 		this.set('previewImage', undefined)
 		if (data && data.type) {
 			const typeCache = await getDefaultClient().typeClient().getTypeCache()
-			const isMedia = typeCache.isTypeInstanceOf(data.type, 'media')
-			if (isMedia) {
-				const media: Video = data as Video
-				const id = media.logo || media.id
+			const logoItem = typeCache.asInstanceOf<FlowrLogoitem>(data, 'flowr.logoitem')
+			if (logoItem) {
```
Comment (lmancel) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]:
```javascript
let id
if (logoItem) {
  id = logoItem.logo || logoItem.id
} else {
  id = data.id
}
```

File: packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts [hunk 11]
```diff
@@ -11,7 +11,7 @@ import './ozone-item-preview.html'
 import * as Config from 'ozone-config'
 
 import { customElement, property } from 'taktik-polymer-typescript'
-import { Item, Video } from 'ozone-type'
+import { Item, FlowrLogoitem } from 'ozone-type'
 import { OzoneMediaUrl, OzonePreviewSize } from 'ozone-media-url'
 import { getDefaultClient } from 'ozone-default-client'
 import '../ozone-item-action/ozone-item-action'
```
Comment (lmancel) [inline packages/ozone-material/ozone-item-preview/src/ozone-item-preview/ozone-item-preview.ts]: Remove `import '../ozone-item-action/ozone-item-action'`

**Pull request merged**